lib/operations/connect.js

```diff
@@ -260,6 +260,18 @@ function connect(mongoClient, url, options, callback) {
     throw new Error('no callback function provided');
   }
 
+  // Has a connection already been established?
+  if (mongoClient.topology) {
+    if (mongoClient.topology.isConnected()) {
+      throw new Error('already connected');
+    }
+    // this is a reconnect, clear out old state
```

Comment:
This might be better to do somewhere inside `close`, just wanted to get some feedback.

Reply:
I think the eventual goal would be that the `topology` contains all of the state, and is removed on `close`. I think it's fine to do additional cleanups of these other types, but agree with you that this should most likely move to `close` or the `close` operation.